traitsui/qt4/button_editor.py

```diff
@@ -153,3 +166,12 @@ def init(self, parent):
         self.sync_value(self.factory.label_value, "label", "from")
         self.control.clicked.connect(self.update_object)
         self.set_tooltip()
+
+    def dispose(self):
+        """ Disposes of the contents of an editor.
+        """
+        if self.control is not None:
+            self.control.clicked.disconnect(self.update_object)
+
+        # enthought/traitsui#884
+        Editor.dispose(self)
```

Comment:
Nit: I would add a `FIXME` to the comment so that it clearly stands out as something to be fixed.